File: leetcode/easy/1614. Maximum Nesting Depth of the Parentheses/solution.js
// https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

/**
 * @param {string} s
 * @return {number}
 */
function maxDepth(s) {
    let max = 0, count = 0;
    for (let i = 0; i < s.length; i += 1) {
        if (s[i] === '(') count += 1;
        if (s[i] === ')') count -= 1;
        max = count > max ? count : max;
    }
    return max;
};

File: leetcode/easy/1614. Maximum Nesting Depth of the Parentheses/solution.cc
// https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/
#include <iostream>
#include <string>

using namespace std;

class Solution {
    public:
        int maxDepth(string s) {
            int res = 0, cnt = 0;

            for (int i = 0; i < s.length(); ++i) {
                if (s[i] == '(') cnt++;
                if (s[i] == ')') cnt--;
                res = max(res, cnt);
            }
            return res;
        };
};

int main() {
    Solution solution;
    int ans = solution.maxDepth("(1)+((2))+(((3)))");
    cout << ans;
    return 0;
}